src/speculators/train/trainer.py

```
@@ -75,23 +77,73 @@ def __init__(
+            p = self._training_state_path(epoch)
+            p.parent.mkdir(parents=True, exist_ok=True)
+            p.write_text(json.dumps(state))
+
+    def _load_training_state(self) -> dict:
+        epoch = self.checkpointer.previous_epoch
+        p = self._training_state_path(epoch)
+        if p.exists():
+            try:
+                return json.loads(p.read_text())
+            except Exception:
+                pass
```

Ruff (0.15.15)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)